app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -1507,13 +1507,14 @@ private void setSyncText(Pair<Long, int[]> syncDetails, String syncTextKey) {
         if (syncTextKey == null) {
             syncTextKey = isDemoUser() ? "home.sync.demo" : "home.sync";
         }
-        if (syncDetails.second[0] > 0) {
-            Spannable syncIndicator = (this.localize("home.sync.indicator", new String[]{String.valueOf(syncDetails.second[0]), Localization.get(syncTextKey)}));
-            syncButton.setNotificationText(syncIndicator);
-            adapter.notifyDataSetChanged();
-
-        } else {
-            if (syncButton != null) syncButton.setText(this.localize(syncTextKey));
+        if (syncButton != null) {
+            if (syncDetails.second[0] > 0) {
+                Spannable syncIndicator = (this.localize("home.sync.indicator", new String[]{String.valueOf(syncDetails.second[0]), Localization.get(syncTextKey)}));
+                syncButton.setNotificationText(syncIndicator);
+                adapter.notifyDataSetChanged();
```

Comment:
Exactly, it only makes sense to set the syncButton text if it has been created (i.e., is visible)